### PR TITLE
drivers: spi: rv32m1: Added LPSPI driver for RV32M1

### PR DESCRIPTION
**Pull request by karstenkoenig**
Modified mcux SPI driver to also support the RV32M1 SOC as they have fairly compatible LPSPI peripherals. They both use their own SOC specific HAL, but those share a fairly common fsl_lpspi interface. Also enabled the SPI flash module on the openVEGA board as a first user of the driver.

### CODE REVIEW

**Comment by karstenkoenig** [comment]
I've followed mostly the lead of this commit by @henrikbrixandersen that added I2C, 
https://github.com/zephyrproject-rtos/zephyr/commit/8a4dbb5b0357b5cbfb1a3d63d65353ecd6cee665

Except that I've extended the mcux spi driver with rv32m1 specifics, which is only adding  CLOCK_SetIpSrc() to select the peripheral clock source.

@MaureenHelm I don't know if extending the mcux driver is ok with you, or if there should rather be a rv32m1 standalone driver. The difference at least in fsl_lpspi.h looks minor, but that might be deceiving

**Comment by karstenkoenig** [comment]
In terms of testing I ran the spi_flash sample on the RV32M1, and that seems to be able to erase + write + readback. I don't have any other board that uses the mcux spi driver so cannot test if that is broken now :-( I built BOARD=twr_ke18f with SPI enabled, and that worked.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thank you @karstenkoenig for the PR. I'm sorry for the delay in my review.

I'd prefer to keep the rv32m1 and mcux drivers separate. Would you mind removing those commits and rebasing please?

**Comment by karstenkoenig** [comment]
@MaureenHelm 
No worries about the delay, thanks for catching it :-)
I dropped combining both drivers inside of the mcux lpspi driver and forked into it's own .c driver. But I otherwise did create a carbon copy, including keeping structures and functions named with mcux. That way patches would transfer more easily. Is this acceptable? 

Also this relies on https://github.com/zephyrproject-rtos/hal_openisa/pull/2

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thanks @karstenkoenig! I tested `samples/drivers/spi_flash` and it works well.

Please update `west.yml` now that zephyrproject-rtos/hal_openisa#2 is merged. I'm concerned that CI didn't fail without this update, but it looks like you did the right thing to make CI pick up the spi flash sample.

File: samples/drivers/spi_flash/boards/rv32m1_vega_ri5cy.conf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#
+# Copyright (c) 2019 Karsten Koenig <karsten.koenig.030@gmail.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+CONFIG_SPI_NOR=y
+CONFIG_SPI_1=y
```
[the remark this answers is not in the document]
Comment (karstenkoenig) [reply samples/drivers/spi_flash/boards/rv32m1_vega_ri5cy.conf]: Done

**Comment by karstenkoenig** [comment]
@MaureenHelm Adjusted the west repo pointer and rebased on top of master due to a codeowner merge conflict

**Comment by karstenkoenig** [comment]
Btw I also added openisa as a vendor, not so sure how that got picked considering it's a NXP datasheet and a Fgreescale marking :-)
I just kept that inline with the prefix that was used for all other rv32m1 drivers

File: soc/riscv/openisa_rv32m1/Kconfig.defconfig [hunk 2]
```diff
@@ -162,6 +162,14 @@ config I2C_RV32M1_LPI2C
 
 endif # I2C
 
+if SPI
+
+config SPI_RV32M1_LPSPI
+	def_bool y
```
Comment (MaureenHelm) [inline soc/riscv/openisa_rv32m1/Kconfig.defconfig]: s/def_bool/default/

File: boards/riscv/rv32m1_vega/Kconfig.defconfig [hunk 3]
```diff
@@ -68,4 +68,11 @@ config I2C_3
 
 endif # I2C
 
+if SPI
+
+config SPI_1
+	def_bool y
```
Comment (MaureenHelm) [inline boards/riscv/rv32m1_vega/Kconfig.defconfig]: s/def_bool/default/

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Sorry, two more small changes. @ulfalizer is actively working to clean up `def_bool`'s

**Review by tbursztyka — APPROVED** [review]
seems fine to me

File: soc/riscv/openisa_rv32m1/Kconfig.defconfig (hunk 2 above)
Comment (karstenkoenig) [reply soc/riscv/openisa_rv32m1/Kconfig.defconfig]: Done, and thanks for looking out for things like that!

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**